### DISCUSSION TITLE
Solar Export

### DISCUSSION THREAD

**Discussion by uknn88** [kellerza/sunsynk#148] [Q&A] [answered]
Hello,

I’ve got a 8kw hybrid single phase sunsynk inverter. I’m currently running this add on in HA.

I’m trying to automate it so it only exports when battery is full. Reason behind this is to stop the 0-50w pull from grid (trickle is set to 0).

after some research, having solar export setting turned off stops it pulling.

so I’m trying to automate ‘solar export’ however this isn’t a switch that appears when I add it as ‘solar_export’ in the configuration.

any clue what I need to do or if it’s even possible? I see ‘solar_export’ in the 3 phase python lib but not single phase.

There is a similar post on April 17th from wevcove and it was suggested to check load limit. However this doesn’t control that particular ‘solar export’ setting.

thanks!

**Comment by kellerza** [comment]
It could be register 247? --> Solar sell -- R/W -- 0x00 solar  Don't sell 0x01 solar sell`


You can try adding a custom sensor - see [mysensors](https://kellerza.github.io/sunsynk/reference/mysensors)
```python
from sunsynk.rwsensors import SwitchRWSensor

SENSORS += SwitchRWSensor(247, "Solar Export")
```

**Comment by wevcore** [comment]
Hello, 

I can confirm that this works - register 247 does switch the solar export on/off

Thanks!

[This is also the answer to the same question I asked a few weeks ago in thread: https://github.com/kellerza/sunsynk/discussions/126]

**Comment by uknn88** [comment]
Hello,Thanks guys for the reply. So register_247 and add it to the config?ThanksSent from my iPhoneOn 18 Jun 2023, at 22:04, wevcore ***@***.***> wrote:﻿
Hello,
I can confirm that this works - register 247 does switch the solar export on/off
[This is also the answer to the same question I asked a few weeks ago in thread: https://github.com//discussions/126]

**Comment by uknn88** [comment]
So I created custom sensor script but doesn’t seem to read it as suggested

**Comment by wevcore** [comment, marked as answer]
So I had to use a mix of @Kellerza's answer and the custom sensor guide.

I have this in the mysensors.py file and it's working for me:

```
from sunsynk import AMPS, CELSIUS, KWH, VOLT, WATT
from sunsynk.rwsensors import NumberRWSensor, SelectRWSensor, TimeRWSensor, SwitchRWSensor
from sunsynk.sensors import (
    MathSensor,
    Sensor,
    SensorDefinitions,
    TempSensor,
)

SENSORS = SensorDefinitions()

SENSORS += SwitchRWSensor(247, "Solar Export")
```

**Reply by iBeech** [reply]
I have put a pull request in to add this sensor: https://github.com/kellerza/sunsynk/pull/196/

**Comment by uknn88** [comment]
Perfect it works! Thanks for the help all!
